### DISCUSSION TITLE
New achievements are counterproductive

### DISCUSSION THREAD

**Discussion by rossjrw** [community/community#18204] [Other Feature Feedback, Questions, & Ideas]
I'm seeing a couple of new achievements in my profile. They are:

* 'YOLO': seemingly for merging pull requests without reviews
* 'Pull Shark': for having opened pull requests that have since been merged.

Having achievements displaying on a user page always risks giving the impression that the user actively sought out those achievements; that they are an 'achievement hunter'; that they endorse and are proud of whatever achievements are displaying. That's not the case with these achievements - they don't reward any special behaviour, only arbitrary numbers and activities.

They don't give the impression that I'm a professional developer, but that I'm a novice coder who's just finished my first project. A potential employer glancing at my profile, perhaps unfamiliar with this very niche GitHub-specific feature, might assume a negative impression of me.

These specific achievements only emphasise my issue. As a developer I don't particularly want to have YOLO and a shark face plastered on my profile - I feel that they might be of more interest to a demographic far younger than me. The achievement names seem to support this - ['Galaxy Brain'](https://github.blog/2022-06-09-introducing-achievements-recognizing-the-many-stages-of-a-developers-coding-journey/) referring to a meme and 'YOLO' being another meme, albeit a decade old. (I don't know if 'Pull Shark' is a reference to anything, but I assume that it is, because it doesn't mean anything to me otherwise.)

But even beyond the theme, these achievements don't encourage positive behaviours. As anyone who's tried to set up KPIs on a bunch of developers will tell you, it's really difficult to measure and reward developer progress numerically. 'Pull Shark' for example incentivises opening pull requests, but not, say, opening *good* or *useful* pull requests (it does require that the pull requests are eventually merged, which is a nice touch). 'YOLO' appears to be a one-off, which is far better than incentivising creating and merging pull requests in a junk repository with reckless abandon - but in that case, what's the point of rewarding that at all?

'Pull Shark' is especially reductive: it displays on my profile as "Pull Shark x 3", which could be read as my having merged 3 pull requests - but clicking into it reveals that I've actually merged at least 128, with the number in the achievement title seemingly being found by $\lfloor \log(4 *n) /( 3 * log(2)) \rfloor$ where $n$ is the PR count - if that's the case, I could have merged anywhere between 128 and 1023 pull requests and this achievement wouldn't care.

To compare these to the two extant achievements that I'm aware of:

* 'Arctic Code Vault Contributor': not a useful achievement, because it's more of an advertisement for something that GitHub is doing than something the user has done specifically. Practically everyone active in open source has it - no personal value
* 'Mars Helicopter 2020 Contributor': describes something discrete and meaningful, and acts as a community rallying point. Really cool. An excellent achievement with genuine personal value - I wish I had it!

Overall, my feedback for the new achievements (those that I'm aware of, anyway), is that they are not particularly useful and even a little harmful. I feel that new achievements should reward specific *achievements* rather than hitting arbitrary quotas.

**Comment by ssb22** [comment]
Likewise, I got the arctic code vault badge and 2 "pull sharks" and I set them to not display.  I might not mind so much if they were displayed less prominently, especially on mobile view where the default place is above your popular or pinned repositories (I think repositories are more important than badges, so I don't want anyone to have to scroll down past badges before they can start to see my repositories).  Still I'm glad that they can be turned off in the profile settings.

**Comment by su77ungr** [comment]
Imagine flexxing your Github achievements on the Twitter ...

![image](https://user-images.githubusercontent.com/69374354/173024291-fa4c5808-6650-40c4-a82e-a2b09ee9ee6e.png)

src: https://github.blog/2022-06-09-introducing-achievements-recognizing-the-many-stages-of-a-developers-coding-journey/

**Reply by slcls** [reply]
Cringe lol

**Reply by trackpadpro** [reply]
I have never even understood why GitHub pushes Twitter specifically. I can think of some better options, but only one is displayable. Do they get a lot of money from Twitter?

**Reply by thenanjay** [reply]
exactly

**Reply by Ysrbolles** [reply]
hahahah

**Comment by jaysonmulwa** [comment]
I totally agree with this.

The **YOLO** badge especially can potentially portray a developer in a negative light - As a rule breaker/ protocol breaker/ someone who doesn't follow processes. This is wrong,  there could be a ton of reasons why something is merged without review. 

It feels authoritarian and not ideal for developers who go above and beyond to have working code deployed fast.
It sucks out the creativity, and blocks out future innovation for the process itself.

GitHub might be shooting itself in the foot here. Will we see alternative code hosting services that are non-opinionated, i.e. don't dictate what the process of authoring and deploying code should be like?.  A revolution like is happening on the web-2  to web-3 sphere?

**Reply by johnhamelink** [reply]
There is no web 3 revolution

**Reply by jaysonmulwa** [reply]
@johnhamelink 

revolution /rɛvəˈluːʃ(ə)n/ _noun_

- A dramatic and wide-reaching change in conditions, attitudes, or operation.

**Reply by johnhamelink** [reply]
@jaysonmulwa exactly: not a top-down marketing push from crypto investors looking for their next pump and dump victims.

**Comment by Neodevils** [comment]
I aggree.

**Comment by kamuridesu** [comment]
I don't know what they plan but it is in beta. They can use these achievements only in beta and add more meaningful badges in the future, at least this is what I'm expecting.

**Reply by Neodevils** [reply]
Yeah, it's a Beta feature though...

**Reply by rossjrw** [reply]
I'm not sure it can really be considered a beta if it's public and opt-out. An A/B test maybe, but I don't have info on that.

**Comment by byronbytes** [comment]
I completely agree with what you said. I feel like that there's been a pattern in recent corporations becoming more... well how do I say it.. "childish"? For example, Discord used to be a decently good chatting platform before they started focusing on Nitro and "communities" instead of what they used to be for. (Mostly because of the pandemic and they could score revenue there), but they ended up modernizing for the bad. 

With the newest improvements with GitHub, I can tell this is where it's going to go. Most people who use GitHub are about over the age of 16, and the designs of, for example, the Pull Shark icon looks like something from cocomelon, and I don't think anyone wants something remotely percieved as childish shown on their profile. 

Also, as I said earlier, if they could manage to do this properly, it would go pretty well. Instead of maybe turning it into achievements they could make it some type of trophy system, that could only be visible to you (or mutual followers) where it would show how many commits, repos, pr's, etc. Something that isn't easy to get and actually worth a trophy.

**Comment by shukkkur** [comment]
I am sure there are definitely more meaningful achievements badges to come, perhaps a useful feature would be to also add an option of selecting which achievement to display and which not.

**Comment by johnhamelink** [comment]
This sort of engagement-brain nonsense is exactly why I'm moving away from Github. That and working with organisations like ICE in the US, which is frankly disgusting.

We need to stop building addiction into our products. It's breaking the internet and destroying people's lives in the process. I'm not exaggerating.

**Reply by KjetilIN** [reply]
Good reasoning. For me it is more about logging progress and experience. 
If the badges actually represents progress or not, i am not sure. I hope they add achievements that reflects what you actually have done over the years.

**Comment by xn4k** [comment]
I mean let people do what they want, if you dont like it, just switch it off lol

**Reply by invisible-defects** [reply]
Exactly. I feel like those things might be targeted towards younger audience, kids who are just making their first steps into programming communities. And you see people who are in their 30s/40s complaining how those cartoonish emblems make no sense all around GH discussions :D

**Reply by hoonsubin** [reply]
For me, it's not that the achievement system is bad. It's the achievements that are stupid. I would love to have great achievements and add that to my CV.

**Reply by xn4k** [reply]
You can clearly see how retarded some github users are, downvoting people's comments just because they wrote they like this option. Funny as hell.

**Comment by ckormanyos** [comment]
> achievements are counterproductive

I just received the Pull Shark, unannounced. I appreciate the achievement. But do not like their superficial characteristics, name or look-and-feel of the emblem.

While I do like getting achievements and this seems like it adds motivation to contribute to open-source, I find the actual way that these achievements are named and how they look to be somewhat off. The pull shark from a distance (I'm kind of old) looked like some kind of a scary ghost, skull or monster. Plus, some contributors (if they figure out the pull shark is actually a shark) might sincerely be afraid of sharks, or similar attributes. I think the symbology of the emblem chosen for this award sends off the wrong vibe.

I'm also sceptical of the one with the brain. What does it mean to not get the brain? Does it mean differentiation regarding cognitive ability? This is too competitive and might be offensive in certain occasions.

I understand that achievements need themes, but is there any way to choose more neutral themes. The pink unicorn whtn GHA is full loaded seems OK for me. Hardly anyone is afraid of unicorns. Maybe ther is another idea with less potential for fear/comparison/competitive negativeness...?

**Reply by Gamebuster19901** [reply]
I don't like the new achievements either, but your reasoning is just terrible. It's a cartoon shark. If someone can't handle seeing it because it genuinely gives them fear, they should not be on github, but instead getting psychiatric help.

Nobody is going to think that because you don't have the brain badge, that you're somehow impaired... It's rewarded for participating in discussions.

I would suggest stop trying to find a way to be offended by everything.

The new achievements give the wrong 'vibe' not for the reasons you state, but because they promote bad practices (merge PR without review, for example), look unprofessional, and are gamifying the platform.

**Comment by WolfyzDBois** [comment]
I agree with what rossjrw said.
It is particularly difficult to see the "level" of certain badges: it is almost the same thing that appears on the profile, whether you have made 1 or 100 commits).
I don't think (at least I hope) that there will be many success hunters.

**Reply by pixeldesu** [reply]
> I don't think (at least I hope) that there will be many success hunters.

There will be achievement hunters, no doubt.

A good example for that, albeit it having a physical reward, is Hacktoberfest. It always had people "cheating" to get the PRs done, but as it got more popular, the "abuse of the system" got so bad that multiple rules got introduced etc.

Whatever kind of reward you can get, people will try to get it _somehow_ with the least effort.

**Comment by marcosbarker** [comment]
I confess that it took me a while to understand what the 'YOLO' badge was, the idea is so absurd. Being "rewarded" for something that is wrong anywhere, imagine this practice at work. It's the "how to get fired in 5 days" badge.

**Comment by Gamebuster19901** [comment]
Imagine having the 2020 Mars Helicopter achievement, but YOLO and Shark are displayed more prominently.
I would be so mad if I had the Mars Badge. I think they need to keep the achievements serious, and stop trying to gamify the platform.

Now I'm going to disable achievements on my profile...

**Comment by hulchenko** [comment]
I am loving this feature. Encourages community to interact with each other and services more than ever!

**Comment by cloudoutloud** [comment]
I love the idea of achievements, it provides motivation for people to contribute to open source and get involved.

Although I do feel they should be harder to obtain.
The 'Pull Shark' achievement should be pull requests based on contributions to a repository that is not your own at least.

**Comment by jef** [comment]
all will probably be okay.

**Comment by samuelfuchs** [comment]
I actually like them. But I like the points you made.

**Comment by Eris-js** [comment]
I usually collect these badges very often. I don't know how to get it

**Comment by maackle** [comment]
I completely agree with the OP. The Mars Helicopter achievement is the only one that's actually cool and useful (and I also wish I had it!) -- the rest is noise, or even counterproductive. I'll be disabling my "achievements" now, I feel more embarrassment than pride in having my work activities over the last decade reduced to a "shark x3" medal.

**Comment by hoonsubin** [comment]
I agree. At least the badges should be something good like how many stars I got, how much I participate in projects and all that.

I hope they don't add paid skins or anything. Or God forbid, NFT profile pics

**Comment by NowDev** [comment]
this seems to be GitHub's attempt to be more like a social network, idk if i agree with that but you can just turn it off, it's not a big issue.

**Comment by jessamynwest** [comment]
I like the idea of achievements and often like how they work other places (Stack Exchange, Reddit) where they can be sort of like longevity/productivity awards. These seem to be an odd combination of things which include non-achievements (Arctic Code Vault - I did nothing to "achieve" this) and, as people said above, badges for things that might not be seen to be positive (YOLO, shark). As [rossjrw](https://github.com/rossjrw) said in the original post, having achievement badges for productive and prosocial community behaviors seems like it would be a worthwhile addition, especially if people can opt out if they hate it. Unfortunately, this isn't that.

**Comment by KjetilIN** [comment]
The **Yolo** achievement seems a bit odd. I like the idea of badges. There should be more of them, but they should indicate how productive you are as a developer. **Pull shark** achievement is okay. I like the idea of logging how many PRs you have done.

What about achievement like: 
- **Active Coder** : Code consistently over a year. Should improve over time. 
- **Reviewer** for reviewing PRs. (Same ass Pull Shark)
- **Multi Tasker**: having to commit to multiple repositories over a period of time 

Just brainstorming here of course, but maybe....

**Comment by Fishrot** [comment]
Absolutely! The FAMAS Mecha Industries is a budget-friendly gem, exuding pure badass vibes. If you're into affordable awesomeness, this sleek option won't break the bank. Trading can be tough, but some swaps are worth it for the rarity. Immerse yourself in the stylish realm of CS:GO skins with [https://www.red-redial.net/how-to-sell-cs/](https://www.red-redial.net/how-to-sell-cs/).

**Comment by github-staff** [comment]
Thanks so much for posting in our Community @rossjrw. Because of an influx of users creating low quality, nonsensical or spam posts in the Community with the aim of earning certain Achievements, we’ve made the difficult decision to disable the ability to earn Achievements in our Community. You can learn more about this decision in our announcement post here [Achievements will no longer be available in the Community](https://github.com/orgs/community/discussions/106536).

Let us know if you have any other questions 🚀

**Reply by rossjrw** [reply]
I'd love to be able to say I'm surprised! Perhaps this merits some reconsideration of the feature as a whole rather than a band-aid solution?

**Reply by jessamynwest** [reply]
Yes. I think this was a feature with some good aspects but just needed to be thought through a little more.
